controllers/provider-openstack/pkg/controller/controlplane/valuesprovider.go

```diff
@@ -0,0 +1,244 @@
+	"github.com/gardener/gardener-extensions/pkg/util"
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils/chart"
+	"github.com/gardener/gardener/pkg/utils/secrets"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener-extensions/controllers/provider-openstack/pkg/internal"
+	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+)
+
+// Object names
+const (
+	cloudControllerManagerDeploymentName = "cloud-controller-manager"
+	cloudControllerManagerServerName     = "cloud-controller-manager-server"
+)
+
+var controlPlaneSecrets = &secrets.Secrets{
+	CertificateSecretConfigs: map[string]*secrets.CertificateSecretConfig{
+		gardencorev1alpha1.SecretNameCACluster: {
+			Name:       gardencorev1alpha1.SecretNameCACluster,
+			CommonName: "kubernetes",
+			CertType:   secrets.CACert,
+		},
+	},
+	SecretConfigsFunc: func(cas map[string]*secrets.Certificate, clusterName string) []secrets.ConfigInterface {
+		return []secrets.ConfigInterface{
+			&secrets.ControlPlaneSecretConfig{
+				CertificateSecretConfig: &secrets.CertificateSecretConfig{
+					Name:         cloudControllerManagerDeploymentName,
+					CommonName:   "system:cloud-controller-manager",
+					Organization: []string{user.SystemPrivilegedGroup},
+					CertType:     secrets.ClientCert,
+					SigningCA:    cas[gardencorev1alpha1.SecretNameCACluster],
+				},
+				KubeConfigRequest: &secrets.KubeConfigRequest{
+					ClusterName:  clusterName,
+					APIServerURL: common.KubeAPIServerDeploymentName,
+				},
+			},
+			&secrets.ControlPlaneSecretConfig{
+				CertificateSecretConfig: &secrets.CertificateSecretConfig{
+					Name:       cloudControllerManagerServerName,
+					CommonName: cloudControllerManagerDeploymentName,
+					DNSNames:   controlplane.DNSNamesForService(cloudControllerManagerDeploymentName, clusterName),
+					CertType:   secrets.ServerCert,
+					SigningCA:  cas[gardencorev1alpha1.SecretNameCACluster],
+				},
+			},
+		}
+	},
+}
+
+var configChart = &chart.Chart{
+	Name: "cloud-provider-config",
+	Path: filepath.Join(openstack.InternalChartsPath, "cloud-provider-config"),
+	Objects: []*chart.Object{
+		{
+			Type: &corev1.ConfigMap{},
+			Name: openstack.CloudProviderConfigName,
+		},
+	},
+}
+
+var ccmChart = &chart.Chart{
+	Name: "cloud-controller-manager",
+	Path: filepath.Join(openstack.InternalChartsPath, "cloud-controller-manager"),
+	// TODO Images: []string{internal.HyperkubeImageName},
```

Comment:
You can add the `hyperkube` image to `images.yaml`, see how this is done for other providers.